cpp/include/Ice/Proxy.h

```diff
@@ -1479,7 +1479,7 @@ ICE_API bool proxyIdentityAndFacetEqual(const ::std::shared_ptr<ObjectPrx>& lhs,
  */
 
 struct ProxyIdentityLess
-#if (ICE_CPLUSPLUS < 201703L)
+#if (ICE_CPLUSPLUS < 201103L)
```

Comment:
I think it would be cleaner from an source compatibility point of view to suppress the warning, instead of changing the API. See #4800.

Reply:
I will rebase after you merge #4800